### PR TITLE
Update Cloudflare/GA analytics tools to latest design/behavior

### PR DESCRIPTION
**Pull request by blackjackkent**
#### Changes proposed in this Pull Request

* This PR refactors the GA/Cloudflare Analytics cards on Marketing -> Traffic to match the designs listed in pbAPfg-1fn-p2. It also adds logic by which GA will be disabled when a new Cloudflare tracking ID gets set, and vice versa.

#### Testing instructions

* Check out this diff to your local Calypso environment.
* Check out D56767-code to your sandbox.
* Navigate to Tools->Marketing->Traffic on a free site account.
* Verify that the Cloudflare and Google Analytics tool options are listed as requiring a premium or higher account.

image

* Navigate to Tools->Marketing->Traffic on a premium account.
*  Verify that you can enable the Cloudflare analytics feature.
* Paste the following snippet into the text field:
```
<!-- Cloudflare Web Analytics -->
<script defer 
    src='https://static.cloudflareinsights.com/beacon.min.js' 
    data-cf-beacon='{"token": "716d4d275266427cb78934b7919e73e6"}'>
</script>
<!-- End Cloudflare Web Analytics -->
```
* Verify that when tabbing out of the field (or otherwise blurring it) this snippet is condensed to only the token value (i.e. `716d4d275266427cb78934b7919e73e6`)
* After saving this update, verify that, when visiting the site, you can see a Cloudflare analytics block in the footer code:
Screen Shot 2021-02-03 at 2 19 47 PM
* Return to Tools->Marketing->Traffic. Verify that you can enable the Google Analytics plugin as well.

Related to 428-gh-Automattic/dotcom-manage

### CODE REVIEW

File: client/my-sites/site-settings/analytics/form-cloudflare-analytics.js [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+				showIcon={ true }
+				title={ nudgeTitle }
+			/>
+		);
+		return (
+			<form id="analytics" onSubmit={ handleSubmitForm }>
+				{ isModalVisible && (
+					<AnalyticsToggleConfirmationDialog
+						productToDisable="Google Analytics"
+						productToEnable="Cloudflare"
```
Comment (a8ci18n) [inline client/my-sites/site-settings/analytics/form-cloudflare-analytics.js]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Cloudflare CDN' )` *ES Score: 6*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/cloudflare-analytics-save-settings)

**Comment by blackjackkent** [comment]
There is currently a bug in this which is causing the logic that triggers the confirmation modal to appear to only work the first time it is run on a given page load. Toggling back again to the other service doesn't trigger it again unless you reload the page first.

I'm pretty baffled as to why this is at the moment so if anyone reviewing this has any idea, please let me know. (It's kind of an edge case though so I'm not sure if it counts as a blocker to merge.)

File: client/my-sites/site-settings/analytics/form-cloudflare-analytics.js [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+		false
+	);
+	const isSubmitButtonDisabled =
+		isRequestingSettings || isSavingSettings || ! isCodeValid || ! enableForm;
+
+	useEffect( () => {
+		if ( fields?.cloudflare_analytics?.code ) {
+			setIsCloudflareEnabled( true );
+		}
+	}, [ fields ] );
```
Comment (gwwar) [inline client/my-sites/site-settings/analytics/form-cloudflare-analytics.js:67]: Maybe try ```[ fields?.cloudflare_analytics?.code ]``` instead here, if we can't trust the array ref to be updated on any state changes

**Comment by kwight** [comment]
If we're stuck on certain behaviour, are there any parts that can be broken out into separate PRs that would be closer to merge? Like the card refactoring?

**Comment by blackjackkent** [comment]
Updated based on discussion in p1612908228011400-slack-C01HV13S295 which removed the complexity of trying to make GA/Cloudflare usage exclusive to each other. Also updated site setting name based on discussion in https://github.com/Automattic/jetpack/pull/18771

File: client/my-sites/site-settings/analytics/form-cloudflare-analytics.js [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+			/>
+		);
+		return (
+			<form id="analytics" onSubmit={ handleSubmitForm }>
+				<SettingsSectionHeader
+					disabled={ isSubmitButtonDisabled }
+					isSaving={ isSavingSettings }
+					onButtonClick={ handleSubmitForm }
+					showButton
+					title={ translate( 'Cloudflare' ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/analytics/form-cloudflare-analytics.js:146]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Cloudflare CDN' )` *ES Score: 6*

**Comment by blackjackkent** [comment]
Updated testing instructions based on changes to D56767-code.

**Comment by kwight** [comment]
I tested this with D56767-code and it worked well, except for the case of pasting in a snippet. When pasting, the form field displays the correct extracted ID, but the settings API call is wrapping the value in an array. Manually entering any valid value sends the correct value (not wrapped in an array).

| Pasted | Manual |
| --- | --- |
| Screen Shot 2021-02-17 at 1 05 28 PM | Screen Shot 2021-02-17 at 1 06 24 PM |

**Comment by mreishus** [comment]
* On a free site, I see the upsell banners linking me to purchase a plan
* On a premium site, I can set my CloudFlare and GA ids
  * Also seeing the same Array bug as Kirk
```
<script defer
	src='https://static.cloudflareinsights.com/beacon.min.js'
	data-cf-beacon='{"token": "Array"}'>
``` 
* Are modals still included?
  * I saw a comment about modals not always working, but I wasn't able to get any modals to display when turning on and off GA/Cloudflare analytics.

**Comment by blackjackkent** [comment]
> Are modals still included?

@mreishus See the discussion in p1612908228011400-slack-C01HV13S295 - we decided to remove that requirement. :)

Good catch (with @kwight ) on the array issue; I will fix that today.

**Comment by blackjackkent** [comment]
Pushed a fix to use the proper key from the regex result when parsing a pasted JS snippet.

**Comment by kwight** [comment]
Looks good – I can see the snippet being added with the key now. I also saw some CORS errors – not sure if they're a problem (that can be followup).

**Review by kwight — APPROVED** [review]
🚢

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/5535772

Thank you @blackjackkent for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/5535772).

**Pull request merged**